packages/shared/src/common/date.ts

```diff
@@ -108,3 +109,66 @@ export function sortDate(
     ? buildDayjs(date1).diff(buildDayjs(date2))
     : buildDayjs(date2).diff(buildDayjs(date1));
 }
+
+export function throwIfInvalidHl7Timestamp(input: string): boolean {
+  if (!input || input.length !== 14) {
+    throw new MetriportError("Invalid HL7 date string format: expected YYYYMMDDHHMMSS");
+  }
+
+  if (!/^\d{14}$/.test(input)) {
+    throw new MetriportError("Invalid HL7 date string: must contain only digits");
+  }
+
+  // Parse YYYYMMDDHHMMSS format
+  const year = input.substring(0, 4);
+  const month = input.substring(4, 6);
+  const day = input.substring(6, 8);
+  const hour = input.substring(8, 10);
+  const minute = input.substring(10, 12);
+  const second = input.substring(12, 14);
+
+  // Validate date components
+  const monthNum = parseInt(month);
+  const dayNum = parseInt(day);
+  const hourNum = parseInt(hour);
+  const minuteNum = parseInt(minute);
+  const secondNum = parseInt(second);
+
+  if (monthNum < 1 || monthNum > 12) {
+    throw new MetriportError("Invalid month in HL7 date string");
+  }
+  if (dayNum < 1 || dayNum > 31) {
+    throw new MetriportError("Invalid day in HL7 date string");
+  }
+  if (hourNum > 23 || minuteNum > 59 || secondNum > 59) {
+    throw new MetriportError("Invalid time in HL7 date string");
+  }
+
+  const dateStr = `${year}-${month}-${day}T${hour}:${minute}:${second}Z`;
+  const parsedDate = buildDayjs(dateStr);
+
+  if (!parsedDate.isValid()) {
+    throw new MetriportError("Invalid HL7 timestamp");
+  }
+
+  return true;
+}
+
+/**
+ * Converts an HL7 timestamp to an ISO 8601 formatted date string.
+ *
+ * @param hl7DateString The HL7 timestamp to convert
+ * @returns ISO 8601 formatted date string
+ */
+export function hl7ToIso8601(hl7DateString: string): string {
```

Comment:
ditto: use a non hl7 specific name for this

Reply:
addressed